src/api/app/controllers/webui/cloud/azure/configurations_controller.rb

```diff
@@ -0,0 +1,51 @@
+module Webui
+  module Cloud
+    module Azure
+      class ConfigurationsController < WebuiController
+        before_action :require_login
+        before_action :set_breadcrumb
+        before_action -> { feature_active?(:cloud_upload) }
+        before_action -> { feature_active?(:cloud_upload_azure) }
+
+        def show
+          @crumb_list << 'Azure Configuration'
+          @azure_configuration = User.current.azure_configuration || User.current.create_azure_configuration
+          @configuration_available = @azure_configuration.application_id && @azure_configuration.application_key
+        end
+
+        def update
+          @azure_configuration = User.current.azure_configuration
+
+          if @azure_configuration.update(permitted_params)
+            flash[:success] = 'Successfully updated your Azure configuration.'
+          else
+            flash[:error] = "Could not update your Azure configuration: #{@azure_configuration.errors.full_messages.to_sentence}."
+          end
+
+          redirect_to cloud_azure_configuration_path
+        end
+
+        def destroy
+          @azure_configuration = User.current.azure_configuration
+          @azure_configuration.destroy! if @azure_configuration
+
+          flash[:success] = "You've successfully deleted your Azure configuration."
+          redirect_to cloud_azure_configuration_path
+        end
+
+        private
+
+        def set_breadcrumb
+          @crumb_list = [
+            WebuiController.helpers.link_to('Cloud Upload', cloud_upload_index_path),
+            WebuiController.helpers.link_to('Configuration', cloud_configuration_index_path)
+          ]
+        end
+
+        def permitted_params
+          params.require(:cloud_azure_configuration).permit(:application_id, :application_key)
+        end
+      end
+    end
+  end
+end
```

Comment:
:thought_balloon: in general, why do you deviate from how scaffold-ed controllers look?

Reply:
I don't understand what you mean.

Reply:
This is a new controller. It doesn't look like the 'default' rails template that `rails scaffold` would create. Not that you should scaffold this but in general it's a good idea to stick to this style, as this is what most people know.

Reply:
It's a copy of the EC2 controller. I don't get your point as this is just a controller with a `#show`, `#update`, and `#destroy` method.

Reply:
Everything is understandable at some point. The question is how fast you reach that point :-)
There are subtle differences like naming the of the params method, how you redirect, how extensible this is, how to apply callbacks etc. etc.

All of these subtle differences I have to decode. If they look like everything looks I don't have to. 

```ruby
class BlahsController < ApplicationController
  # PATCH/PUT /blahs/1
  def update
    respond_to do |format|
      if @blah.update(blah_params)
        format.html { redirect_to @blah, notice: 'Blah was successfully updated.' }
      else
        format.html { render :edit }
      end
    end
  end

...

private
    # Use callbacks to share common setup or constraints between actions.
    def set_blah
      @blah = Blah.find(params[:id])
    end

    # Never trust parameters from the scary internet, only allow the white list through.
    def blah_params
      params.fetch(:blah, {})
    end
end

Reply:
So and what do you want to change now? That we should use `respond_to` ? I don't really see the difference to our controller. We're using `permitted_params` as you do, the `.find` improvement you already suggested, and as for the redirect, you want to do the suggestion to provide the `flash` as a parameter?

Reply:
What I'm talking about is style, not about functional changes...